controllers/model/utils.go

```diff
@@ -48,3 +48,12 @@ func SetCommonLabels(obj metav1.ObjectMetaAccessor) {
 	}
 	meta.SetLabels(labels)
 }
+
+func SetInheritedLabels(obj metav1.ObjectMetaAccessor, extraLabels map[string]string) {
+	SetCommonLabels(obj)
+	meta := obj.GetObjectMeta()
+	labels := meta.GetLabels()
+	for k, v := range extraLabels {
+		labels[k] = v
+	}
+}
```

Comment:
Since `SetCommonLabels` is now only invoked from the `SetInheritedLabels` function, I would consider merging them.
Especially since `SetInheritedLabels` does not explicitly call `meta.SetLabels(labels)` which feels quite ambiguous 😅 
```suggestion
func SetInheritedLabels(obj metav1.ObjectMetaAccessor, instanceLabels map[string]string) {
	meta := obj.GetObjectMeta()
	labels := meta.GetLabels()

	// Ensure default CommonLabels for child resources
	if labels == nil {
		labels = CommonLabels
	} else {
		for k, v := range CommonLabels {
			labels[k] = v
		}
	}

	// Inherit labels from the parent grafana instance if any
	for k, v := range instanceLabels {
		labels[k] = v
	}

	meta.SetLabels(labels)
}
```

Otherwise, I cannot seem to break it and it seems to work from my limited testing :D

Reply:
this should be ok now